modules/javafx.media/src/main/native/jfxmedia/platform/osx/avf/AVFMediaPlayer.mm

```diff
@@ -653,10 +653,8 @@ - (void) sendPixelBuffer:(CVPixelBufferRef)buf frameTime:(double)frameTime hostT
 
 - (void) sendSpectrumEventDuration:(double)duration timestamp:(double)timestamp {
     if (eventHandler) {
-        if (timestamp < 0) {
-            timestamp = self.currentTime;
-        }
-        eventHandler->SendAudioSpectrumEvent(timestamp, duration);
+        // Always true for queryTimestamp do to JDK-8240694
```

Comment:
that should be "due" to... you might also mention that we need to query on the event thread to avoid a deadlock.